### ISSUE TITLE
"Oops, I couldn't figure out how to switch to root." with non-`install.sh` name.

### ISSUE THREAD

**Issue by cben** [sandstorm-io/sandstorm#631]
This is really minor but anyway:
With both full server and dev paths:
```
To set up Sandstorm, we will need to use sudo.
Sandstorm's database and web interface won't run as root.
OK to continue? [yes] yes
Oops, I couldn't figure out how to switch to root. Please re-run the installer 
as root.
```
Strange, I'm on Ubuntu 15.04, `sudo` works perfectly.
Aha:
```
  elif [ "$(basename $SCRIPT_NAME)" == install.sh ] && [ -e "$0" ]; then
```
I wanted to peek at the script before I run it (was TLDR so runned anyway ;-)
but I was lazy and saved as `s.sh` not `install.sh`...

**Comment by kentonv** [comment]
Yeah the problem is not that it couldn't figure out how to sudo, but that it couldn't figure out how to run itself inside sudo.

The reason it checks the script name is because I found it difficult to reliably distinguish different ways that install.sh could have been executed (e.g. via pipe, via "./install.sh", via "bash install.sh", or any number of other ways). I figured if $0 was install.sh then I could be reasonably certain that the user had downloaded the script and rut it locally. Since the worst case was that the user would have to run sudo themselves, I figured it wasn't a big deal to cover everything.

But up to @paulproteus if he wants to make this heuristic better, or more informative.
